src/test/java/com/snowflake/kafka/connector/internal/InternalStageIT.java

```diff
@@ -12,86 +12,123 @@
-            stageName2, "appName/tableName/partition/testCacheFileName" + i, "Any cache");
-      }
-      List<String> files2 = service.listStage(stageName2, "appName/tableName/partition/testCache");
-      assert files2.size() == fileNumber;
-      System.out.println(
-          Logging.logMessage("Time: {} ms", (System.currentTimeMillis() - startTime)));
+    // PUT two files to stageName1
+    startTime = System.currentTimeMillis();
+    agent.putWithCache(stageName1, "testCacheFileName1", "Any cache", stageType);
+    agent.putWithCache(stageName1, "testCacheFileName2", "Any cache", stageType);
+    List<String> files1 = service.listStage(stageName1, "testCache");
+    assert files1.size() == 2;
+    System.out.println(Logging.logMessage("Time: {} ms", (System.currentTimeMillis() - startTime)));
+
+    // PUT 50 files to stageName2
+    startTime = System.currentTimeMillis();
+    for (int i = 0; i < fileNumber; i++) {
+      agent.putWithCache(
+          stageName2, "appName/tableName/partition/testCacheFileName" + i, "Any cache", stageType);
     }
+    List<String> files2 = service.listStage(stageName2, "appName/tableName/partition/testCache");
+    assert files2.size() == fileNumber;
+    System.out.println(Logging.logMessage("Time: {} ms", (System.currentTimeMillis() - startTime)));
+  }
+
+  @Ignore
+  @Test
+  public void testComparePutAPIVersionsForGCS() {
 
-    // When stage is GCS
-    if (!usePutWithCacheApi(agent, stageName1)) {
-      // PUT 50 files to stageName3
-      startTime = System.currentTimeMillis();
-      for (int i = 0; i < fileNumber; i++) {
-        service.put(stageName3, "appName/tableName/partition/testNoCacheFileName" + i, "Any cache");
-      }
-      List<String> files3 =
-          service.listStage(stageName3, "appName/tableName/partition/testNoCache");
-      assert files3.size() == fileNumber;
-      System.out.println(
-          Logging.logMessage("Time: {} ms", (System.currentTimeMillis() - startTime)));
+    // PUT 50 files to stageName3
+    final int numberOfFiles = 50;
+    long startTime = System.currentTimeMillis();
+    for (int i = 0; i < numberOfFiles; i++) {
+      service.put(
+          stageNameGCSPut, "appName/tableName/partition/testNoCacheFileName" + i, "Any data");
+    }
+    System.out.println(
+        Logging.logMessage(
+            "Time for putting {} files in GCS is:{} ms",
+            numberOfFiles,
+            (System.currentTimeMillis() - startTime)));
+
+    List<String> files =
+        service.listStage(stageNameGCSPut, "appName/tableName/partition/testNoCacheFileName");
+    assert files.size() == numberOfFiles;
+
+    System.out.println("Same test but with uploadWithoutConnection API");
+    startTime = System.currentTimeMillis();
+    for (int i = 0; i < numberOfFiles; i++) {
+      service.putWithCache(
+          stageNameGCSPutWithCache,
+          "appName/tableName/partition/testCacheFileName" + i,
+          "Any data");
     }
+    System.out.println(
```

Comment:
How long does putWithCache take vs put for GCS? How does it stack up with other clouds? Can we add the expected range as a comment?

Reply:
I found uploadWithoutConnection to be overall faster compared to put. Will comments. For other clouds it will be faster since we cache the creds. (Only one call in 30 mins + expired credentials exception)